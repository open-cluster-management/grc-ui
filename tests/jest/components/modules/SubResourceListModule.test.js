/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import ResourceOverviewModule from '../../../../src-web/components/modules/SubResourceListModule'
import
{ policieSubResourceListTestingResourceData,
  policieSubResourceListTestingItemData,
  clusterSubResourceListTestingResourceData,
  clusterSubResourceListTestingItemData,
} from './ModuleTestingData'

describe('PolicieSubResourceList view', () => {
  const staticResourceData = policieSubResourceListTestingResourceData
  const items = policieSubResourceListTestingItemData
  const linkFixedName = {2 : {fixedName:'table.actions.launch.cluster', urlTarget:'_blank'}}
  it('renders as expected', () => {
    const component = renderer.create(
      <BrowserRouter>
        <ResourceOverviewModule
          staticResourceData={staticResourceData}
          items={items}
          listSubItems={true}
          linkFixedName={linkFixedName}
        /></BrowserRouter>)
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('ClusterSubResourceList view', () => {
  const staticResourceData = clusterSubResourceListTestingResourceData
  const items = clusterSubResourceListTestingItemData
  it('renders as expected', () => {
    const component = renderer.create(
      <BrowserRouter>
        <ResourceOverviewModule
          staticResourceData={staticResourceData}
          items={items}
          listSubItems={true} /></BrowserRouter>)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
