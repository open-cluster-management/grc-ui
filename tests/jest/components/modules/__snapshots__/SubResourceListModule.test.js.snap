// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterSubResourceList view renders as expected 1`] = `
<div
  className="bx--module bx--module--double structured-list-module"
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__content"
    >
      <table
        className="bx--data-table-v2 resource-table-main simple-table bx--data-table-v2--zebra"
      >
        <thead>
          <tr>
            <th
              className="bx--header-index-0"
              scope="col"
            />
            <th
              className="bx--header-index-1"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Policy name
              </span>
            </th>
            <th
              className="bx--header-index-2"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Violations
              </span>
            </th>
            <th
              className="bx--header-index-3"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Control
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="bx--parent-row-v2"
            data-parent-row={true}
          >
            <td
              className="bx--table-expand-v2"
              headers="expand"
            >
              <button
                aria-label="Overflow-menu"
                className="bx--table-expand-v2__button"
                onClick={[Function]}
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-expand-v2__svg"
                  fillRule="evenodd"
                  height="12"
                  role="img"
                  viewBox="0 0 7 12"
                  width="7"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                  />
                </svg>
              </button>
            </td>
            <td>
              <p>
                policy-ma
              </p>
            </td>
            <td>
              <p>
                1
              </p>
            </td>
            <td>
              <p>
                MutationAdvisor
              </p>
            </td>
          </tr>
          <tr
            className="bx--parent-row-v2"
            data-parent-row={true}
          >
            <td
              className="bx--table-expand-v2"
              headers="expand"
            >
              <button
                aria-label="Overflow-menu"
                className="bx--table-expand-v2__button"
                onClick={[Function]}
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-expand-v2__svg"
                  fillRule="evenodd"
                  height="12"
                  role="img"
                  viewBox="0 0 7 12"
                  width="7"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                  />
                </svg>
              </button>
            </td>
            <td>
              <p>
                policy-pod
              </p>
            </td>
            <td>
              <p>
                1
              </p>
            </td>
            <td>
              <p>
                MutationAdvisor,VA
              </p>
            </td>
          </tr>
          <tr
            className="bx--parent-row-v2"
            data-parent-row={true}
          >
            <td
              className="bx--table-expand-v2"
              headers="expand"
            >
              <button
                aria-label="Overflow-menu"
                className="bx--table-expand-v2__button"
                onClick={[Function]}
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-expand-v2__svg"
                  fillRule="evenodd"
                  height="12"
                  role="img"
                  viewBox="0 0 7 12"
                  width="7"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                  />
                </svg>
              </button>
            </td>
            <td>
              <p>
                policy-role
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                MutationAdvisor,VA
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`PolicieSubResourceList view renders as expected 1`] = `
<div
  className="bx--module bx--module--double structured-list-module"
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__content"
    >
      <table
        className="bx--data-table-v2 resource-table-main simple-table bx--data-table-v2--zebra"
      >
        <thead>
          <tr>
            <th
              className="bx--header-index-0"
              scope="col"
            />
            <th
              className="bx--header-index-1"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Cluster name
              </span>
            </th>
            <th
              className="bx--header-index-2"
              scope="col"
            >
              <span
                className="bx--table-header-label"
              >
                Violations
              </span>
            </th>
            <th
              className="bx--header-index-3"
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr
            className="bx--parent-row-v2"
            data-parent-row={true}
          >
            <td
              className="bx--table-expand-v2"
              headers="expand"
            >
              <button
                aria-label="Overflow-menu"
                className="bx--table-expand-v2__button"
                onClick={[Function]}
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-expand-v2__svg"
                  fillRule="evenodd"
                  height="12"
                  role="img"
                  viewBox="0 0 7 12"
                  width="7"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                  />
                </svg>
              </button>
            </td>
            <td>
              <p>
                cluster1
              </p>
            </td>
            <td>
              <p>
                1
              </p>
            </td>
            <td>
              <p>
                <a
                  className="bx--link"
                  href="/https://9.42.78.148:8443"
                  onClick={[Function]}
                  target="_blank"
                >
                  Launch cluster
                </a>
              </p>
            </td>
          </tr>
          <tr
            className="bx--parent-row-v2"
            data-parent-row={true}
          >
            <td
              className="bx--table-expand-v2"
              headers="expand"
            >
              <button
                aria-label="Overflow-menu"
                className="bx--table-expand-v2__button"
                onClick={[Function]}
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-expand-v2__svg"
                  fillRule="evenodd"
                  height="12"
                  role="img"
                  viewBox="0 0 7 12"
                  width="7"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
                  />
                </svg>
              </button>
            </td>
            <td>
              <p>
                clusterhub
              </p>
            </td>
            <td>
              <p>
                0
              </p>
            </td>
            <td>
              <p>
                <a
                  className="bx--link"
                  href="/https://9.42.81.137:8443"
                  onClick={[Function]}
                  target="_blank"
                >
                  Launch cluster
                </a>
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
